cpp/src/utilities/nvtx/nvtx_utils.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "nvtx_utils.hpp"
+#include "utilities/error_utils.hpp"
+
+#ifdef USE_NVTX
+#include <nvToolsExt.h>
+#endif
+
+namespace cudf {
+namespace nvtx {
+
+void range_push(const char* name, color color)
+{
+  range_push_hex(name, static_cast<uint32_t>(color));
+}
+
+void range_push_hex(const char* name, uint32_t color)
+{
+  CUDF_EXPECTS(name != nullptr, "Null name string.");
```

Comment:
Maybe put inside `USE_NVTX` so we don't pay for the check if it's disabled?

Reply:
Ok, will do. I was thinking it might be safer to always fail on invalid input even if USE_NVTX is disabled, so that a potential bug doesn't hide until someone enables it.

Reply:
good point.